File: ts/80_Remove_Duplicates_from_Sorted_Array_II.ts
// https://leetcode.com/problems/remove-duplicates-from-sorted-array-ii/description/
function removeDuplicatesII(nums: number[]): number {
    const chk: Record<string, number> = {};
    let idx = 0;
    for (let i = 0; i < nums.length; i++) {
        if (!(nums[i] in chk)) {
            chk[nums[i]] = 1;
            nums[idx++] = nums[i];
        } else if (chk[nums[i]] >= 2) {
            continue;
        } else {
            chk[nums[i]] += 1;
            nums[idx++] = nums[i];
        }
    }
    return idx;
};

File: C++/80_Remove_Duplicates_from_Sorted_Array_II.cpp
// https://leetcode.com/problems/remove-duplicates-from-sorted-array-ii/description/
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int chk =1;
        for (int i=1;i<nums.size();i++){
            if (chk==1 || nums[i] != nums[chk-2]){
                nums[chk++] = nums[i];
            }
        }
        return chk;
    }
};